packages/mui-material/src/Backdrop/Backdrop.js

```diff
@@ -67,22 +68,38 @@ const Backdrop = React.forwardRef(function Backdrop(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const rootSlotProps = slotProps.root ?? componentsProps.root;
+  const backwardCompatibleSlots = {
+    transition: TransitionComponentProp,
+    root: components.Root,
+    ...slots,
+  };
+  const backwardCompatibleSlotProps = { ...componentsProps, ...slotProps };
+  const externalForwardedProps = {
+    slots: backwardCompatibleSlots,
+    slotProps: backwardCompatibleSlotProps,
+  };
+
+  const [RootSlot, rootProps] = useSlot('root', {
+    elementType: BackdropRoot,
+    externalForwardedProps,
+    className: clsx(classes.root, className),
+    ownerState,
+  });
+
+  const [TransitionSlot, transitionProps] = useSlot('transition', {
+    elementType: Fade,
+    externalForwardedProps,
+    ownerState,
+  });
+
+  delete transitionProps.ownerState;
```

Comment:
This is no longer needed: https://github.com/mui/material-ui/pull/41187

We can remove it

Reply:
Oh, ok, my bad. Yeah that makes sense. Let's leave it.